scripts/scil_volume_pairwise_comparison.py

```diff
@@ -0,0 +1,206 @@
+                             add_overwrite_arg,
+                             add_processes_arg,
+                             add_reference_arg,
+                             add_verbose_arg,
+                             assert_headers_compatible,
+                             assert_inputs_exist,
+                             assert_outputs_exist,
+                             validate_nbr_processes)
+from scilpy.image.labels import get_data_as_labels
+from scilpy.tractanalysis.reproducibility_measures import compute_dice_voxel, \
+    compute_bundle_adjacency_voxel
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_volumes', nargs='+',
+                   help='Path of the input volumes.')
+    p.add_argument('out_json',
+                   help='Path of the output json file.')
+
+    p.add_argument('--adjency_no_overlap', action='store_true',
+                   help='If set, do not count zeros in the average BA.')
+
+    p.add_argument('--single_compare',
+                   help='Compare inputs to this single file.')
+    p.add_argument('--ratio', action='store_true',
+                   help='Compute overlap and overreach as a ratio over the '
+                        'reference volume.\nCan only be used if also using '
+                        '--single_compare`.')
+
+    add_processes_arg(p)
+    add_reference_arg(p)
+    add_json_args(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def compute_all_measures(args):
+    filename_1 = args[0][0]
+    filename_2 = args[0][1]
+    adjency_no_overlap = args[1]
+    ratio = args[2]
+
+    img_1, dtype_1 = load_img(filename_1)
+
+    if np.issubdtype(dtype_1, np.floating):
+        raise ValueError('Input file {} is of type float.\n'
+                         'Please convert to uint8 (mask) or '
+                         'uint16 (labels).'.format(filename_1))
+    img_2, dtype_2 = load_img(filename_2)
+    if np.issubdtype(dtype_2, np.floating):
+        raise ValueError('Input file {} is of type float.\n'
+                         'Please convert to uint8 (mask) or '
+                         'uint16 (labels).'.format(filename_2))
+
+    data_1 = get_data_as_labels(img_1)
+    data_2 = get_data_as_labels(img_2)
+
+    _, _, voxel_size, _ = get_reference_info(img_1)
+    voxel_size = np.product(voxel_size)
+
+    # Exclude 0 (background)
+    unique_values_1 = np.unique(data_1)[1:]
+    unique_values_2 = np.unique(data_2)[1:]
+    union_values = np.union1d(unique_values_1, unique_values_2)
+
+    dict_measures = {}
+    for val in union_values:
+        binary_1 = np.zeros(data_1.shape, dtype=np.uint8)
+        binary_1[data_1 == val] = 1
+        binary_2 = np.zeros(data_2.shape, dtype=np.uint8)
+        binary_2[data_2 == val] = 1
+
+        # These measures are in mm^3
+        volume_overlap = np.count_nonzero(binary_1 * binary_2)
```

Comment:
See if you could use:
scilpy.tractanalysis.scoring.compute_f1_overlap_overreach, which can be tested with unit tests.

Reply:
Since this does not always use <ratio>, it is not always normalized between 0 - 1 (this script is not for GT). So I will leave it like that.